File: ios/PocketOracle/Features/Home/DailyRitualModels.swift
import Foundation

enum DailyRitualDestination: String {
    case tarot
    case astrology
    case bazi

    var icon: String {
        switch self {
        case .tarot: return "moon.stars.fill"
        case .astrology: return "sparkles"
        case .bazi: return "leaf.fill"
        }
    }

    var localizedTitle: String {
        switch self {
        case .tarot:
            return AppLanguageOption.isChinese ? "抽一张今日塔罗" : "Draw Today's Tarot"
        case .astrology:
            return AppLanguageOption.isChinese ? "看看今日行运" : "Read Today's Transit"
        case .bazi:
            return AppLanguageOption.isChinese ? "整理今日能量" : "Review Today's Energy"
        }
    }
}

struct DailyTarotHighlight {
    let card: TarotCard
    let isUpright: Bool
    let title: String
    let insight: String
    let keywords: [String]
}

struct DailyTransitHighlight {
    let title: String
    let summary: String
    let signName: String?
    let needsProfileCompletion: Bool
}

struct DailyRitual {
    let date: Date
    let quote: DailyQuote
    let ritualTitle: String
    let ritualSummary: String
    let destination: DailyRitualDestination
    let destinationReason: String
    let energyTitle: String
    let energyBody: String
    let tarot: DailyTarotHighlight
    let transit: DailyTransitHighlight
    let promptTitle: String
    let promptBody: String
}

File: android/feature/feature-home/src/main/kotlin/com/fortunepocket/feature/home/DailyRitualModels.kt
package com.fortunepocket.feature.home

import com.fortunepocket.core.model.DailyQuote
import com.fortunepocket.core.model.TarotCard
import com.fortunepocket.core.model.ZodiacSign

data class RitualCopy(
    val zh: String,
    val en: String
) {
    fun resolve(isZh: Boolean): String = if (isZh) zh else en
}

enum class DailyRitualDestination {
    TAROT,
    ASTROLOGY,
    BAZI;

    fun title(isZh: Boolean): String = when (this) {
        TAROT -> if (isZh) "抽塔罗" else "Tarot Reading"
        ASTROLOGY -> if (isZh) "本命盘占星" else "Natal Chart"
        BAZI -> if (isZh) "八字分析" else "Bazi Analysis"
    }
}

data class DailyTarotHighlight(
    val title: RitualCopy,
    val card: TarotCard,
    val isUpright: Boolean,
    val insight: RitualCopy
)

data class DailyTransitHighlight(
    val title: RitualCopy,
    val sign: ZodiacSign?,
    val summary: RitualCopy,
    val needsBirthday: Boolean
)

data class DailyRitual(
    val generatedAtMillis: Long,
    val ritualTitle: RitualCopy,
    val ritualSummary: RitualCopy,
    val destination: DailyRitualDestination,
    val destinationReason: RitualCopy,
    val energyTitle: RitualCopy,
    val energyBody: RitualCopy,
    val tarot: DailyTarotHighlight,
    val transit: DailyTransitHighlight,
    val promptTitle: RitualCopy,
    val promptBody: RitualCopy,
    val quote: DailyQuote
)
